fvm/environment/event_emitter.go

```diff
@@ -295,8 +290,8 @@ func (collection *EventCollection) TotalEventCounter() uint32 {
 	return collection.eventCounter
 }
 
-// IsServiceEvent determines whether or not an emitted Cadence event is
-// considered a service event for the given chain.
+// IsServiceEvent determines whether an emitted Cadence event is considered a service event for the given chain.
+// An event is a service event if it is defined in the `systemcontracts` package allow-list.
```

Comment:
I think this is a significant change. Would suggest to be explicit 
```suggestion
// An event is a service event if it is defined in the `systemcontracts` package allow-list.
// Note that we have *removed* the prior constraint that service events can only be 
// emitted in the system chunk. Now a system smart contract can emit service events
// as part of any transaction.
```